Change EcommerceService's is_enabled to accept User instead of request

vkaracic opened the pull request:
And change the verification link in the sidebar to redirect to new basket if the EcommerceService is enabled.
This is a fix for https://openedx.atlassian.net/browse/SOL-1712
@mjfrey @pbaruah

edx-webhook commented:
Thanks for the pull request, @vkaracic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?number=11852&repo=edx%2Fedx-platform

douglashall commented on lms/djangoapps/courseware/date_summary.py:
```diff
@@ -204,6 +205,9 @@ class VerifiedUpgradeDeadlineDate(DateSummary):
+        ecommerce_service = EcommerceService()
+        if ecommerce_service.is_enabled(self.user):
+            return ecommerce_service.payment_page_url()
         return reverse('verify_student_upgrade_and_verify', args=(self.course.id,))
```
Do we need a similar check [here](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/verify_student/views.py#L483)?

vkaracic replied:
That part of the code should be investigated but it isn't part of this ticket so I've created a new ticket for it: https://openedx.atlassian.net/browse/SOL-1714

douglashall commented on lms/djangoapps/courseware/date_summary.py:
```diff
@@ -204,6 +205,9 @@ class VerifiedUpgradeDeadlineDate(DateSummary):
     @property
     def link(self):
+        ecommerce_service = EcommerceService()
+        if ecommerce_service.is_enabled(self.user):
```
Do we need a test to cover this condition?

mattdrayer replied:
It could just be that it is late, but I'm not seeing 'user' defined in this class or the DateSummary parent -- where is it coming from?

vkaracic replied:
Get some sleep, Matt :)
https://github.com/edx/edx-platform/blob/vkaracic/SOL-1712/lms/djangoapps/courseware/date_summary.py#L68

mattdrayer replied:
Haha, okay, fair enough :)

douglashall commented:
@vkaracic Couple of questions inline. We should set this up on a sandbox for testing if it isn't already.

vkaracic commented:
I've added a test for the implementation. I also changed the link to redirect to ``single-item`` view together with the course SKU, so that a user adds that course to her basket, previously an error would appear that the SKU is missing.

vkaracic commented:
jenkins run bokchoy

douglashall commented:
:+1:

mjfrey commented:
:+1:

edx-webhook commented:
Thanks for the pull request, @vkaracic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-platform&number=11852
